StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1600ElementsMustBeDocumented.cs

```diff
@@ -44,7 +50,195 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+        private void HandlePropertyDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            PropertyDeclarationSyntax declaration = context.Node as PropertyDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                { 
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.Identifier.GetLocation()));
+                }
+            }
+        }
+
+
+
+        private void HandleIndexerDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            IndexerDeclarationSyntax declaration = context.Node as IndexerDeclarationSyntax;
+
+            if (declaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, declaration.ThisKeyword.GetLocation()));
+                }
+            }
+        }
+
+        private void HandleFieldDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            FieldDeclarationSyntax declaration = context.Node as FieldDeclarationSyntax;
+            var variableDeclaration = declaration?.Declaration;
+
+            if (variableDeclaration != null && NeedsComment(declaration.Modifiers, SyntaxKind.PrivateKeyword))
+            {
+                var leadingTrivia = declaration.GetLeadingTrivia();
+                var commentTrivia = leadingTrivia.FirstOrDefault(x => x.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia));
+                if (XmlCommentHelper.IsMissingOrEmpty(commentTrivia))
+                {
+                    var locations = variableDeclaration.Variables.Select(v => v.Identifier.GetLocation()).ToArray();
+                    if (locations.Length > 0)
+                    {
+                        context.ReportDiagnostic(Diagnostic.Create(Descriptor, locations.First()));
```

Comment:
I believe the warning should be reported for each field, as opposed to just the first one.